x-pack/plugins/security_solution/public/timelines/containers/index.tsx

```diff
@@ -511,8 +514,24 @@ export const useTimelineEvents = ({
 
   useEffect(() => {
     if (!timelineSearchHandler) return;
+
+    const startTime = performance.now();
     timelineSearchHandler(onTimelineSearchComplete);
-  }, [timelineSearchHandler, onTimelineSearchComplete]);
+    const endTime = performance.now();
+
+    telemetry.reportTimelinesTrackQueryRun({
+      tab: timelineTab ?? undefined,
+      duration: endTime - startTime,
+      language,
+    });
+  }, [
+    timelineSearchHandler,
+    onTimelineSearchComplete,
+    timelineResponse.events,
```

Comment:
`timelineResponse.events` should not be in the dependency array. This is causing infinite loop in this hook.